### PR TITLE
Decoupled map into several smaller modules

### PR DESCRIPTION
**Pull request by Hoolean**
See #36, #33

### CODE REVIEW

File: map.py [hunk 1]
```diff
@@ -1,93 +1,72 @@
...
+
+        # convert the pixels into coordinates
+        relative_x = uncentred_x // TILE_PIX_WIDTH
+        relative_y = uncentred_y // TILE_PIX_HEIGHT
+
+        # make the coordinates absolute, not relative to the player
+        map_x = relative_x + self.centre_player.x
+        map_y = relative_y + self.centre_player.y
+
+        # round down
```
Comment (ob6160) [inline map.py]: remove me :imp:

File: player.py [hunk 2]
```diff
@@ -7,6 +7,7 @@
 from pygame.rect import Rect
 
 import client
+from map import TILE_PIX_WIDTH, TILE_PIX_HEIGHT
```
Comment (ob6160) [inline player.py]: Investigate a better way of including these?

File: player.py [hunk 3]
```diff
@@ -19,19 +20,18 @@ class PlayerException(Exception):
...
         self.is_centre = False
-        self.size = size
-        self.step = self.size[0]
+        self.size = (TILE_PIX_WIDTH, TILE_PIX_HEIGHT)
+        self.step = 1
         self.colour = colour
-        self.name = name
+        self.name = 'Name'
 
-        if len(position) > 0:
```
Comment (ob6160) [inline player.py]: This needs thinking about - it might break multiplayer?

File: tile.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+class TileType(Enum):
+    DIRT = (2, [])
+    BRICK = (7, [ TileAttribute.COLLIDE ])
+
+    def __init__(self, tileset_id, attributes):
+        self.tileset_id = tileset_id
+        self.attributes = 0
+        for attribute in attributes:
+            self.attributes = self.attributes | attribute.value
```
Comment (ob6160) [inline tile.py]: Add comments/separate description about bitwise.

File: tile.py [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        self.surfaces = {}
+
+    def get_surface(self, id):
+        if id in self.surfaces:
+            return self.surfaces[id]
+
+        clip_x, clip_y = self.find_position(id)
+
+        cur_tile_width = (self.image.get_width() / self.grid_dimensions[0])
+        cur_tile_height = (self.image.get_height() / self.grid_dimensions[0])
```
Comment (ob6160) [inline tile.py]: [1]

File: player.py (hunk 3 above)
Comment (Hoolean) [inline player.py]: As of present, the only ill-effect should be that the player momentarily appears at `( 0, 0 )` until their position is updated.

**Pull request merged**